bundle/hash_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package bundle
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+)
+
+func TestHashFile(t *testing.T) {
+
+	tests := map[string]struct {
+		a         string
+		b         string
+		algorithm HashingAlgorithm
+		equal     bool
+	}{
+		"string":      {`"abc"`, `"abc"`, SHA256, true},
+		"map":         {`{"a": "foo", "b": "bar"}`, `{"b": "bar", "a": "foo"}`, MD5, true},
+		"map_2":       {`{"a": {"foo": [1, 2, 3]}, "b": "bar"}`, `{"b": "bar", "a": {"foo": [1, 2, 3]}}`, MD5, true},
+		"map_unequal": {`{"a": {"foo": [1, 2, 3]}, "b": "bar"}`, `{"b": "bars", "a": {"foo": [1, 2, 3]}}`, SHA384, false},
+		"array":       {`[1, 2, 3]`, `[1, 2, 3]`, SHA512, true},
+		"null":        {`null`, `null`, SHA512224, true},
+		"bool":        {`false`, `false`, SHA512256, true},
```

Comment:
I'd think you can transform this into test that checks the hash computed directly from the raw bytes equals to the one the alg computes. Then a few more corner cases to cover in the types of docs hashed: []byte, string with unicode characters, string that could HTML escaped (but won't).

Reply:
Updated with more tests.